### PR TITLE
[serializerFragment] avoid generating duplicate methods

### PR DESCRIPTION
**Pull request by meysholdt**
this suppresses the symptoms of 
https://bugs.eclipse.org/bugs/show_bug.cgi?id=482840 and
https://bugs.eclipse.org/bugs/show_bug.cgi?id=479400

Signed-off-by: Moritz Eysholdt <moritz.eysholdt@itemis.de>

### CODE REVIEW

**Comment by meysholdt** [comment]
Jenkins test

File: plugins/org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/serializer/SerializerFragment2.xtend [hunk 1]
```diff
@@ -227,7 +229,16 @@ class SerializerFragment2 extends AbstractStubGeneratingFragment {
 				�genMethodCreateSequence�
 				
 				�FOR c : newLocalConstraints.sort�
-					�genMethodSequence(c)�
+					�IF methodSignatures.add(c.simpleName -> c.type)�
+						�genMethodSequence(c)�
+					�ELSE�
+						�LOG.warn("Skipped generating duplicate method in " + clazz.simpleName)�
+						�genMethodSequenceComment(c)�
```
Comment (szarnekow) [inline plugins/org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/serializer/SerializerFragment2.xtend]: Won't this become (part of) the javadoc of the next method?

File: plugins/org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/serializer/SerializerFragment2.xtend [hunk 2]
```diff
@@ -227,7 +229,16 @@ class SerializerFragment2 extends AbstractStubGeneratingFragment {
...
+						// This method is uncommented because it has the same signature as another method in this class.
```
Comment (szarnekow) [inline plugins/org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/serializer/SerializerFragment2.xtend]: uncommented?

Comment (meysholdt) [inline plugins/org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/serializer/SerializerFragment2.xtend]: fixed.

File: plugins/org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/serializer/SerializerFragment2.xtend (hunk 1 above)
Comment (meysholdt) [inline plugins/org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/serializer/SerializerFragment2.xtend]: fixed.

**Comment by meysholdt** [comment]
Jenkins test

**Comment by meysholdt** [comment]
Jenkins xtext-runtime-tests-master-latest

**Pull request merged**